diff --git a/arch/riscv/boot/dts/microchip/mpfs-icicle-kit.dts b/arch/riscv/boot/dts/microchip/mpfs-icicle-kit.dts
index c247ed8a3..9257d8fcc 100644
--- a/arch/riscv/boot/dts/microchip/mpfs-icicle-kit.dts
+++ b/arch/riscv/boot/dts/microchip/mpfs-icicle-kit.dts
@@ -19,7 +19,7 @@ / {
 	soc {
 		dma-ranges = <0x14 0x0 0x0 0x80000000 0x0 0x4000000>,
 			     <0x14 0x4000000 0x0 0xc4000000 0x0 0x6000000>,
-			     <0x14 0xa000000 0x0 0x8a000000 0x0 0x8000000>,
+			     <0x14 0xa000000 0x0 0x8a000000 0x0 0x1000000>,
 			     <0x14 0x12000000 0x14 0x12000000 0x0 0x10000000>,
 			     <0x14 0x22000000 0x10 0x22000000 0x0 0x5e000000>;
 	};
@@ -48,7 +48,7 @@ kernel: memory@80000000 {
 
 	ddr_cached_low: memory@8a000000 {
 		device_type = "memory";
-		reg = <0x0 0x8a000000 0x0 0x8000000>;
+		reg = <0x0 0x8a000000 0x0 0x1000000>;
 	};
 
 	ddr_non_cached_low: memory@c4000000 {
@@ -93,37 +93,22 @@ dma_non_cached_high: non-cached-high-buffer {
 			alloc-ranges = <0x14 0x12000000 0x0 0x10000000>;
 		};
 
-		fabricbuf0ddrc: buffer@88000000 {
-			compatible = "shared-dma-pool";
-			reg = <0x0 0x88000000 0x0 0x2000000>;
-			no-map;
-		};
-
 		fabricbuf1ddrnc: buffer@c8000000 {
 			compatible = "shared-dma-pool";
 			reg = <0x0 0xc8000000 0x0 0x2000000>;
 			no-map;
 		};
 
-		fabricbuf2ddrncwcb: buffer@d8000000 {
+		fabricbuf2ddrnc: buffer@cb000000 {
 			compatible = "shared-dma-pool";
-			reg = <0x0 0xd8000000 0x0 0x2000000>;
+			reg = <0x0 0xcb000000 0x0 0x4000000>;
 			no-map;
 		};
 	};
 
-	udmabuf0 {
-		compatible = "ikwzm,u-dma-buf";
-		device-name = "udmabuf-ddr-c0";
-		minor-number = <0>;
-		size = <0x0 0x2000000>;
-		memory-region = <&fabricbuf0ddrc>;
-		sync-mode = <3>;
-	};
-
 	udmabuf1 {
 		compatible = "ikwzm,u-dma-buf";
-		device-name = "udmabuf-ddr-nc0";
+		device-name = "udmabuf-ddr-fabric";
 		minor-number = <1>;
 		size = <0x0 0x2000000>;
 		memory-region = <&fabricbuf1ddrnc>;
@@ -132,12 +117,13 @@ udmabuf1 {
 
 	udmabuf2 {
 		compatible = "ikwzm,u-dma-buf";
-		device-name = "udmabuf-ddr-nc-wcb0";
+		device-name = "udmabuf-ddr-vbx";
 		minor-number = <2>;
-		size = <0x0 0x2000000>;
-		memory-region = <&fabricbuf2ddrncwcb>;
+		size = <0x0 0x4000000>;
+		memory-region = <&fabricbuf2ddrnc>;
 		sync-mode = <3>;
 	};
+
 };
 
 /*
